includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -449,14 +469,18 @@ public function generate_payment_request( $order, $prepared_source ) {
 			];
 		}
 
-		$post_data['metadata'] = apply_filters( 'wc_stripe_payment_metadata', $metadata, $order, $prepared_source );
+		$post_data['metadata'] = apply_filters( 'wc_stripe_payment_metadata', $metadata, $order, $payment_method );
```

Comment:
This may be a BC break since any existing filters will be expecting a $prepared_source object which seems to be slightly different. Same problem below for the filter 'wc_stripe_generate_payment_request'. Probably $prepared_source and $payment_method are similar enough it won't break any integrations. Do you know what properties are different between them? Ideally we'd write a function and convert $payment_method to $prepared_source when one of these filters is registered but I'm guessing it's not really possible. Probably we just need to clearly document the break in the changelog.

Reply:
Good point! I didn't really bother to check what that `$prepared_source` structure was: it basically looks like [this](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/abstracts/abstract-wc-stripe-payment-gateway.php#L750-L755) below.
```
(object) [
    'token_id' => $wc_token_id,
    'customer' => $customer_id,
    'source' => $source_id,
    'source_object' => $source_object,
];
```
In d2693f17632bb962979ce816b4b140b2295de70f, I created a similar object for a payment method [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/d2693f17632bb962979ce816b4b140b2295de70f/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php#L745-L752) and passed it into `generate_payment_request` to keep these filters consistent.